commit 67259e451d5d58b4842776c5696a66d74e157609
Author: Niels Dossche <7771979+nielsdos@users.noreply.github.com>
Date:   Fri Jul 5 23:34:09 2024 +0200

    Fix GH-14834: Error installing PHP when --with-pear is used
    
    libxml2 2.13 makes changes to how the parsing state is set, update our
    code accordingly. In particular, it started reporting entities within
    attributes, while it should only report entities inside text nodes.
    
    Closes GH-14837.

--- a/ext/xml/compat.c
+++ b/ext/xml/compat.c
@@ -376,7 +376,7 @@ _get_entity(void *user, const xmlChar *n
 			ret = xmlGetDocEntity(parser->parser->myDoc, name);
 
 		ZEND_DIAGNOSTIC_IGNORED_START("-Wdeprecated-declarations")
-		if (ret == NULL || (parser->parser->instate != XML_PARSER_ENTITY_VALUE && parser->parser->instate != XML_PARSER_ATTRIBUTE_VALUE)) {
+		if (ret == NULL || parser->parser->instate == XML_PARSER_CONTENT) {
 		ZEND_DIAGNOSTIC_IGNORED_END
 			if (ret == NULL || ret->etype == XML_INTERNAL_GENERAL_ENTITY || ret->etype == XML_INTERNAL_PARAMETER_ENTITY || ret->etype == XML_INTERNAL_PREDEFINED_ENTITY) {
 				/* Predefined entities will expand unless no cdata handler is present */
